openscm/core/views.py

```diff
@@ -303,64 +315,46 @@ def __init__(  # pylint: disable=too-many-arguments
         """
         super().__init__(parameter)
         self._unit_converter = UnitConverter(cast(str, parameter.unit), unit)
-        self._timeseries_converter = TimeseriesConverter(
-            parameter.time_points,
-            time_points,
-            timeseries_type,
-            interpolation_type,
-            extrapolation_type,
-        )
+        self._timeseries_converter = None
+        self._timeseries_type = timeseries_type
+        self._interpolation_type = interpolation_type
+        self._extrapolation_type = extrapolation_type
         self._data = None
         self._locked = False
         self._timeseries = None
+        self._version = 0
         self._writable = False
 
-        def get_data_views_for_children_or_parameter(
-            parameter: _Parameter
-        ) -> Sequence["TimeseriesView"]:
-            if parameter.children:
-                return sum(
-                    (
-                        get_data_views_for_children_or_parameter(p)
-                        for p in parameter.children.values()
-                    ),
-                    [],
-                )
-            return [
-                TimeseriesView(
-                    parameter,
-                    self._unit_converter.target,
-                    self._timeseries_converter._target,  # pylint: disable=protected-access
-                    timeseries_type,
-                    interpolation_type,
-                    extrapolation_type,
-                )
-            ]
-
-        if self._parameter.children:
-            self._child_data_views = get_data_views_for_children_or_parameter(
-                self._parameter
-            )
+        self.time_points = time_points
 
     def _read(self):
-        if self._data is None:
+        if self._data is None or self._version != self._parameter.version:
```

Comment:
A new `TimeseriesConverter` should only be created if the `time_points` in the parameter changed

Reply:
Doing to on every version change is a big performance hit and goes contrary to the idea of the converters and views.

Reply:
Yep nice